src/Dennis/Link/Checker/Analyzer.php

```diff
@@ -194,6 +194,11 @@ public function getInfo($url) {
    * @throws ResourceFailException
    */
   protected function doInfoRequest($url) {
+    // If url is protocol neutral, force it to use https.
```

Comment:
Oof, I'm not keen on this; you would be better to use http and follow any redirect to https. There are still (some) http-only sites out there which will error if you request https.